stats.go

```diff
@@ -438,6 +568,44 @@ func newHistogramBucket(
 	return bucket
 }
 
+type metricExpiryTracker struct {
+	lastUpdateUnix int64
+	expired        uint32
+	scope          *scope
+	name           string
+}
+
+// expiredAndUpdated checks the metric is expired and if so
+// will return true only for the first routine that updates
+// an expired metric, signalling to insert the metric back
+// into the parent scope.
+func (m *metricExpiryTracker) expiredAndUpdated() bool {
+	// We perform a LoadUint32 first for performance reasons
+	if atomic.LoadUint32(&m.expired) == 1 && m.scope != nil &&
```

Comment:
why do we check for `m.scope != nil`, i don't see where we `nil` out the scope, and i think all the constructos set it?